### ISSUE TITLE
Editor freeze if delete script

### ISSUE THREAD

**Issue by RuanLucasGD** [FlaxEngine/FlaxEngine#1773]
**Issue description:**
If you add a script on an Actor with some reference fields, setup and save the scene and after delete the script your editor freeze and cannot reload anything, you cannot reopen your scene too.

**Steps to reproduce:**
* Add this script to an actor:
```
using System;
using System.Collections.Generic;
using FlaxEngine;

namespace Game;
public class MyScript : Script
{
    public Script a;
    public Script b;
    public Actor c;
    public Tag d;
}

```
* Setup all fields with some value on your actor.
* Save the scene.
* Delete the script.
* Editor freeze.

**Flax version:**
1.7 master

**Comment by mafiesto4** [comment]
Should be fixed 806590d1c3350136ae6d0043fa22e06a3faf7002

**Comment by RuanLucasGD** [comment]
@mafiesto4 yeah, this worked for me, thanks!
